### PR TITLE
Environment variables

### PR DESCRIPTION
**Pull request by gahissy**
When working with Docker, passing environment variables is more practical than using a configuration file.
This PR changed the `Config` class to look for an environment variable before checking the pre-configured configuration files.
There is an UPPERCASE_SNAKE rule applied to properties names.
Ex:
forward.url --> FORWARD_URL
database.driver --> DATABASE_DRIVER

### CODE REVIEW

File: src/org/traccar/Config.java [hunk 1]
```diff
@@ -15,48 +15,55 @@
  */
 package org.traccar;
 
+import org.apache.commons.lang3.StringUtils;
```
Comment (tananaev) [inline src/org/traccar/Config.java]: I think we should avoid using this indirect dependency, especially if it's easy to use standard Java.

File: src/org/traccar/Config.java [hunk 2]
```diff
@@ -15,48 +15,55 @@
...
 
         String defaultConfigFile = properties.getProperty("config.default");
         if (defaultConfigFile != null) {
             try (InputStream inputStream = new FileInputStream(defaultConfigFile)) {
-                defaultProperties = new Properties();
+                Properties defaultProperties = new Properties();
                 defaultProperties.loadFromXML(inputStream);
+
+                Enumeration props = defaultProperties.propertyNames();
+                while (props.hasMoreElements()) {
```
Comment (tananaev) [inline src/org/traccar/Config.java]: Do we really need this? I don't like that there are two ways of setting up environment variables.

File: src/org/traccar/Config.java [hunk 3]
```diff
@@ -15,48 +15,55 @@
...
-        return properties.containsKey(key) || defaultProperties != null && defaultProperties.containsKey(key);
+        return properties.containsKey(key);
     }
 
     public String getString(String key) {
-        if (properties.containsKey(key) || defaultProperties == null) {
-            return properties.getProperty(key);
-        } else {
-            return defaultProperties.getProperty(key);
+        String envName = key.toUpperCase().replaceAll("\\.", "_");
```
Comment (tananaev) [inline src/org/traccar/Config.java]: Replacing using regular expressions is a pretty heavy operation. I think we should have some global flag to tell server to consider environment variables. Otherwise just use config.

Also, it would be nice to add underscore between words in camel-case parameters. For example `database.ignoreUnknown` should be converted to `DATABASE_IGNORE_UNKNOWN`.

File: src/org/traccar/Config.java (hunk 2 above)
Comment (gahissy) [inline src/org/traccar/Config.java]: Actually that part is not related to environment variables.
Instead of holding properties into two separate variables `properties` and `defaultProperties` the idea was to merge everything and use only `properties`. That way

Comment (tananaev) [inline src/org/traccar/Config.java]: Sorry, my bad. Totally missed that. Then I think it's a good idea to keep it this way. Is there a better way to merge properties? Properties is a hashtable, so you should be able to just call `putAll` or something like that.

**Comment by gahissy** [comment]
PR updated @tananaev 
- `database.ignoreUnknown` is translated to `DATABASE_IGNORE_UNKNOWN` now
- Test added
- `config.useEnv` (property) and `CONFIG_USE_ENV` (env var) can be used to enable environment variables interpolation
- Default properties and main properties are merged with a cleaner code

**Comment by tananaev** [comment]
I have fixed a problem and did some cleanup including renaming. The problem was the you used properties before loading them.
